### PR TITLE
JBPM-8984 Default runtime strategy of 'New Deployment Unit' wizard sh…

### PR DESCRIPTION
**Pull request by bxf12315**
…ould respect kie-deployment-descriptor.xml when manually inputting GAV

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 1]
```diff
@@ -176,4 +209,51 @@ public void testClose() {
...
+        PageResponse<JarListPageRow> response = new PageResponse<JarListPageRow>();
+        JarListPageRow jarListPageRow = new JarListPageRow();
+        GAV gav = new GAV("test", "test", "");
+        jarListPageRow.setGav(gav);
+        jarListPageRow.setPath("test_path");
+        response.setPageRowList(Arrays.asList(jarListPageRow));
+        when(m2RepoService.listArtifacts(any())).thenReturn(response);
+        when(newContainerFormPresenter.getCurrentGAV()).thenReturn(gav);
+
+        newContainerWizard.pages.add(initWizardPage());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]: mock(WizardPage.class) ?

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/resources/org/kie/workbench/common/screens/server/management/client/resources/i18n/Constants.properties [hunk 2]
```diff
@@ -147,3 +147,4 @@ ClientContainerRuntimeOperation.StopScanner=Stop Scanner
 ContainerUpdateEvent.ErrorMessage=The '{0}' operation has failed in {1} server instance(s).
 ContainerUpdateEvent.WarnMessage=The '{0}' operation may have failed in some of the {1} server instance(s).
 ContainerUpdateEvent.SuccessMessage=The '{0}' operation has succeeded, {1} server instances were affected.
+NewContainer.GAVNotExist=not exist.
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/resources/org/kie/workbench/common/screens/server/management/client/resources/i18n/Constants.properties]: Perhaps use something like: GAV '{0}' not found in the Maven repository

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/resources/org/kie/workbench/common/screens/server/management/client/resources/i18n/Constants.properties]: But if I use java.text.MessageFormat, gwt compile will failed.

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java (hunk 1 above)
Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]: ok

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/resources/org/kie/workbench/common/screens/server/management/client/resources/i18n/Constants.properties (hunk 2 above)
Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/resources/org/kie/workbench/common/screens/server/management/client/resources/i18n/Constants.properties]: done

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by cristianonicolai** [comment]
@bxf12315 the expand feature on the diagram is supposed to go full screen mode.  Ithink we would need more time to look into that, as that is not a crucial feature, my suggestion is to create a specific JIRA only for that.

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java [hunk 3]
```diff
@@ -49,6 +60,10 @@ public boolean equals( final Object o ) {
         if ( !context.equals( that.context ) ) {
             return false;
         }
+
+        if (!isSelected == that.isSelected) {
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]:
```suggestion
        if (isSelected != that.isSelected) {
```

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]: done

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

**Comment by jstastny-cz** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
@cristianonicolai sonar check fail in code Coverage. please review it.

**Comment by cristianonicolai** [comment]
@bxf12315 can you look in the sonar feedback please? looks like you need to increase the test coverage

**Comment by nmirasch** [comment]
@bxf12315 It works fine until you select one of the available GAV from list. It seems there is a problem with the generated DependencyPathSelectedEvent because since that moment, there's no way to see it working again until you reload the page

**Comment by bxf12315** [comment]
> @bxf12315 It works fine until you select one of the available GAV from list. It seems there is a problem with the generated DependencyPathSelectedEvent because since that moment, there's no way to see it working again until you reload the page

yes, when you selected a kjar, will throw DependencyPathSelectedEvent, container config page will refresh. I think, if you select a kjar, it's mean, it is new kjar. what do you think?

**Comment by bxf12315** [comment]
@cristianonicolai @nmirasch updated, please review again.

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by nmirasch** [comment]
@bxf12315 I think it's weird reload the first found kjar when the gav is introduced manually every time you change the page, regarding if you change the run strategy value and select 'pervious' and 'next again you lose the change. For me the expected is that the found gav is loaded the first time you change the page, and then only reload the conf if the manual gav is changed again. This allows change the conf as  if it was a selected one form the available ones.

**Comment by bxf12315** [comment]
> @bxf12315 I think it's weird reload the first found kjar when the gav is introduced manually every time you change the page, regarding if you change the run strategy value and select 'pervious' and 'next again you lose the change. For me the expected is that the found gav is loaded the first time you change the page, and then only reload the conf if the manual gav is changed again. This allows change the conf as if it was a selected one form the available ones.

done

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java [hunk 4]
```diff
@@ -109,30 +129,66 @@ public void close() {
...
+            GAV gav = newContainerFormPresenter.getCurrentGAV();
+            JarListPageRequest request = new JarListPageRequest(0, Integer.MAX_VALUE, null, Arrays.asList("jar"), null, false);
+
+            m2RepoService.call(response -> {
+                List<JarListPageRow> jarListPageRowList = ((PageResponse<JarListPageRow>) response).getPageRowList();
+                Optional<JarListPageRow> optionalJarListPageRow = jarListPageRowList.stream().filter(jarListPageRow -> jarListPageRow.getGav().equals(gav))
+                        .sorted(Comparator.comparing(JarListPageRow::getLastModified).reversed()).findFirst();
+                if (optionalJarListPageRow.isPresent()) {
+                    if (optionalJarListPageRow.get().getPath() != null && !optionalJarListPageRow.get().getPath().isEmpty()) {
+                        dependencyPathSelectedEvent.fire(new DependencyPathSelectedEvent(this, optionalJarListPageRow.get().getPath(), false));
```
Comment (nmirasch) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java]: @bxf12315  If we launch here DependencyPathSelectedEvent with the param isselected = true allows to respect the changes done at Runtime Strategy configuration changed by the user throw the wizard. I think the behaviour would be more consistent with the one that is performed when a kjar is selected from list. WDYT?

**Comment by jstastny-cz** [comment]
Checked the fix in full downstream, seems fine. Both selection from list and manual input reflect the runtime strategy setting of the project.

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java (hunk 4 above)
Comment (cristianonicolai) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java]: +1

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java [hunk 5]
```diff
@@ -20,13 +20,20 @@
 
     private final Object context;
     private final String path;
+    private boolean isSelected = true;
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]: check if needed

Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]: done

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java [hunk 6]
```diff
@@ -109,30 +128,66 @@ public void close() {
...
+            GAV gav = newContainerFormPresenter.getCurrentGAV();
+            JarListPageRequest request = new JarListPageRequest(0, Integer.MAX_VALUE, null, Arrays.asList("jar"), null, false);
+
+            m2RepoService.call(response -> {
+                List<JarListPageRow> jarListPageRowList = ((PageResponse<JarListPageRow>) response).getPageRowList();
+                Optional<JarListPageRow> optionalJarListPageRow = jarListPageRowList.stream().filter(jarListPageRow -> jarListPageRow.getGav().equals(gav))
+                        .sorted(Comparator.comparing(JarListPageRow::getLastModified).reversed()).findFirst();
+                if (optionalJarListPageRow.isPresent()) {
+                    if (optionalJarListPageRow.get().getPath() != null && !optionalJarListPageRow.get().getPath().isEmpty()) {
+                        dependencyPathSelectedEvent.fire(new DependencyPathSelectedEvent(this, optionalJarListPageRow.get().getPath()));
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java]:
```suggestion
                        this.isSelected = true;
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java [hunk 7]
```diff
@@ -48,23 +58,31 @@
...
-                               final ProcessConfigPagePresenter processConfigPagePresenter,
-                               final Caller<SpecManagementService> specManagementService,
-                               final Event<NotificationEvent> notification,
-                               final Event<ServerTemplateSelected> serverTemplateSelectedEvent ) {
+    public NewContainerWizard(final NewContainerFormPresenter newContainerFormPresenter,
+                              final ProcessConfigPagePresenter processConfigPagePresenter,
+                              final Caller<SpecManagementService> specManagementService,
+                              final Event<NotificationEvent> notification,
+                              final Event<ServerTemplateSelected> serverTemplateSelectedEvent,
+                              final Event<DependencyPathSelectedEvent> dependencyPathSelectedEvent,
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java]: Remove dependencyPathSelectedEvent

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java [hunk 8]
```diff
@@ -48,23 +58,31 @@
     private final Event<NotificationEvent> notification;
     private final Event<ServerTemplateSelected> serverTemplateSelectedEvent;
     private ServerTemplate serverTemplate;
+    private Event<DependencyPathSelectedEvent> dependencyPathSelectedEvent;
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizard.java]: remove dependencyPathSelectedEvent

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 9]
```diff
@@ -169,11 +200,153 @@ public void testComplete() {
...
+        response.setPageRowList(Arrays.asList(jarListPageRow));
+        when(m2RepoService.listArtifacts(any())).thenReturn(response);
+        when(newContainerFormPresenter.getCurrentGAV()).thenReturn(gav);
+
+        newContainerWizard.pages.add(mock(WizardPage.class));
+        newContainerWizard.pages.add(mock(WizardPage.class));
+
+        newContainerWizard.pageSelected(1);
+
+        verify(dependencyPathSelectedEvent, never()).fire(any());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]:
```suggestion
        assertFalse(newContainerWizard.isSelected);
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 10]
```diff
@@ -169,11 +200,153 @@ public void testComplete() {
...
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]:
```suggestion
        assertFalse(newContainerWizard.isSelected);
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 11]
```diff
@@ -169,11 +200,153 @@ public void testComplete() {
...
+        newContainerWizard.pages.add(mock(WizardPage.class));
+        newContainerWizard.pages.add(mock(WizardPage.class));
+    }
+
+    @Test
+    public void testPageSelectedPageOne() {
+        preparePageSelected();
+
+        newContainerWizard.pageSelected(0);
+        verify(dependencyPathSelectedEvent, never()).fire(any());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]:
```suggestion
        assertFalse(newContainerWizard.isSelected);
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 12]
```diff
@@ -169,11 +200,153 @@ public void testComplete() {
...
+        verify(notification, never()).fire(any());
+    }
+
+    @Test
+    public void testPageSelectedIsSelected() {
+        preparePageSelected();
+        newContainerWizard.isSelected = true;
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]:
```suggestion
        newContainerWizard.onDependencyPathSelectedEvent(new DependencyPathSelectedEvent("null", "test"));
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 13]
```diff
@@ -169,11 +200,153 @@ public void testComplete() {
...
+        newContainerWizard.pageSelected(1);
+        verify(dependencyPathSelectedEvent, never()).fire(any());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]:
```suggestion
        assertTrue(newContainerWizard.isSelected);
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 14]
```diff
@@ -169,11 +200,153 @@ public void testComplete() {
...
     }
 
-}
\ No newline at end of file
+    @Test
+    public void testPageSelectedPageTwo() {
+        preparePageSelected();
+
+        newContainerWizard.pageSelected(1);
+        verify(dependencyPathSelectedEvent).fire(any());
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]:
```suggestion
        assertTrue(newContainerWizard.isSelected);
```

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 15]
```diff
@@ -76,13 +93,16 @@ public void setup() throws IllegalAccessException, NoSuchFieldException {
...
+        m2RepoServiceCaller = new CallerMock<>(m2RepoService);
         when( newContainerFormPresenter.getView() ).thenReturn( newContainerFormPresenterView );
         newContainerWizard = spy( new NewContainerWizard(
                 newContainerFormPresenter,
                 processConfigPagePresenter,
                 specManagementServiceCaller,
                 notification,
-                serverTemplateSelectedEvent
+                serverTemplateSelectedEvent,
+                dependencyPathSelectedEvent,
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]: remove dependencyPathSelectedEvent;

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java [hunk 16]
```diff
@@ -60,12 +69,20 @@
...
+    M2RepoService m2RepoService;
+
     @Spy
     Event<NotificationEvent> notification = new EventSourceMock<NotificationEvent>();
 
     @Spy
     Event<ServerTemplateSelected> serverTemplateSelectedEvent = new EventSourceMock<ServerTemplateSelected>();
 
+    @Mock
+    Event<DependencyPathSelectedEvent> dependencyPathSelectedEvent = new EventSourceMock<DependencyPathSelectedEvent>();
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/wizard/NewContainerWizardTest.java]: remove dependencyPathSelectedEvent

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java [hunk 17]
```diff
@@ -27,6 +27,7 @@ public DependencyPathSelectedEvent( final Object context,
         this.path = path;
     }
 
+
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]: remove change

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java [hunk 18]
```diff
@@ -35,6 +36,7 @@ public String getPath() {
         return path;
     }
 
+
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]: remove change

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java [hunk 19]
```diff
@@ -49,6 +51,7 @@ public boolean equals( final Object o ) {
         if ( !context.equals( that.context ) ) {
             return false;
         }
+
```
Comment (cristianonicolai) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/events/DependencyPathSelectedEvent.java]: remove change

**Comment by bxf12315** [comment]
Jenkins execute full downstream build

**Pull request merged**